src/plugins/intel_gpu/tests/test_cases/eye.cpp

```diff
@@ -0,0 +1,163 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <algorithm>
+#include <intel_gpu/primitives/data.hpp>
+#include <intel_gpu/primitives/eye.hpp>
+#include <intel_gpu/primitives/input_layout.hpp>
+#include <random>
+#include <vector>
+
+#include "test_utils.h"
+
+using namespace cldnn;
+using namespace ::tests;
+
+namespace {
+
+template <typename T>
+class EyeTest : public ::testing::Test {
+public:
+    void TearDown() override {
+        topology topology;
+
+        topology.add(data("num_rows", num_rows_));
+        topology.add(data("num_columns", num_coloms_));
+        topology.add(data("diagonal_index", diagonal_index_));
+        std::vector<primitive_id> inputs;
+        if (batch_) {
+            topology.add(data("batch", batch_));
+            inputs = {"num_rows", "num_columns", "diagonal_index", "batch"};
+        } else {
+            inputs = {"num_rows", "num_columns", "diagonal_index"};
+        }
+        auto eye_primitive = eye("eye", inputs, tensor{output_shape_});
+        eye_primitive.output_data_type = this->output_type_;
+        eye_primitive.shift = this->diagonal_val;
+        topology.add(std::move(eye_primitive));
+
+        network network(engine_, topology);
+
+        auto outputs = network.execute();
+
+        EXPECT_EQ(outputs.size(), size_t(1));
+        EXPECT_EQ(outputs.begin()->first, "eye");
+
+        auto output = outputs.at("eye").get_memory();
+
+        cldnn::mem_lock<T> output_ptr(output, get_test_stream());
+
+        ASSERT_EQ(output_ptr.size(), expected_output_.size());
+        for (size_t i = 0; i < output_ptr.size(); ++i)
+            EXPECT_TRUE(are_equal(expected_output_[i], output_ptr[i], 2e-3));
+    }
+
+    data_types DataType() const;
+
+protected:
+    engine& engine_ = get_test_engine();
+    memory::ptr num_rows_;
+    memory::ptr num_coloms_;
+    memory::ptr diagonal_index_;
+    memory::ptr batch_;
+    data_types output_type_{DataType()};
+    std::vector<std::int32_t> output_shape_;
+    int32_t diagonal_val;
+    std::vector<T> expected_output_;
+};
+
+template <>
+data_types EyeTest<float>::DataType() const {
+    return data_types::f32;
+}
+
+template <>
+data_types EyeTest<int32_t>::DataType() const {
+    return data_types::i32;
+}
+
+template <>
+data_types EyeTest<int64_t>::DataType() const {
+    return data_types::i64;
+}
+
+template <>
+data_types EyeTest<int8_t>::DataType() const {
+    return data_types::i8;
+}
+using testing::Types;
+typedef Types<float, int32_t, int64_t, int8_t> DataTypes;
+TYPED_TEST_SUITE(EyeTest, DataTypes);
+
+TYPED_TEST(EyeTest, bfyx_1_1_3_2) {
```

Comment:
using the test logic in teardown looks like a hack a little bit. If all the your tests have the same logic and only input's are different I think better approach is to inherit from TestWithParam and separate data in suite (as example you can see gather_gpu_test, reverse_gpu_test. In such case, you will have the advantage to use combine in some cases (for example if you are going to support layouts)
In other case, I see your concept is already used in some other tests in this folder, so please take my account as a suggestion only, not as a request to rework. We can leave it as it is here

Reply:
I took the slice unit test as a reference. Reworked as you proposed